ext/standard/proc_open.c

```diff
@@ -568,61 +812,20 @@ PHP_FUNCTION(proc_open)
 
 #ifdef PHP_WIN32
 	if (other_options) {
-		zval *item = zend_hash_str_find(Z_ARRVAL_P(other_options), "suppress_errors", sizeof("suppress_errors") - 1);
-		if (item != NULL) {
-			if (Z_TYPE_P(item) == IS_TRUE || ((Z_TYPE_P(item) == IS_LONG) && Z_LVAL_P(item))) {
-				suppress_errors = 1;
-			}
-		}
-
+		set_option(other_options, "suppress_errors",      &suppress_errors);
```

Comment:
I think get_option would be the more appropriate name. I'm also wondering why you pass the target variable by reference, rather than using the return value?

Reply:
Both observations are very right. Amended.